code/modules/overmap/exoplanets/decor/turfs/water.dm

```diff
@@ -1,12 +1,109 @@
 /turf/simulated/floor/exoplanet/water
 	does_footprint = FALSE
+	footstep_sound = /singleton/sound_category/water_footstep
+	movement_cost = 4
+	///How many objects are currently on this turf? Used to stop empty water turfs from processing.
+	var/numobjects = 0
+	///Is this water deep enough to drown in?
+	var/deep = TRUE
+	///The overlay used to make it look like atoms on the turf are underwater.
+	var/obj/effect/water_effect/water_overlay
 
 /turf/simulated/floor/exoplanet/water/update_icon()
 	return
 
 /turf/simulated/floor/exoplanet/water/update_dirt()
 	return	// Water doesn't become dirty
 
+/turf/simulated/floor/exoplanet/water/Initialize()
+	. = ..()
+	if(deep)
+		var/obj/effect/water_effect/W = new /obj/effect/water_effect(src)
+		W.icon = icon
+		W.icon_state = icon_state
+		water_overlay = W
+		W.alpha = 128
+	create_reagents(4)
+
+/turf/simulated/floor/exoplanet/water/Destroy()
+	if(water_overlay)
+		qdel(water_overlay)
+		water_overlay = null
+	return ..()
+
+/turf/simulated/floor/exoplanet/water/return_air_for_internal_lifeform(var/mob/living/carbon/L)
+	if(!L)
+		return
+	if(L.lying || deep) //are they lying down/is the water deep enough to keep their head above it?
+		if(water_overlay && L.layer > water_overlay.layer) //are they on a vehicle or something else that physically puts them above the water?
```

Comment:
One problem I saw in the PV events is that water doesn't account for things like rod paths or catwalks above it, so people stepping on a catwalk over sewer would water would just instantly die and not know why. Needs to be addressed if it hasn't